### ISSUE TITLE
--txpool-notify propsal

### ISSUE THREAD

**Issue by wojtasss** [monero-project/monero#6775]
It would be nice to have similar notify as --tx-notify in wallet but wtih new pending transactions in txpool, is possible to implement it?

**Comment by ndorf** [comment]
I agree this could be convenient. 

You could also use the [new ZeroMQ pub/sub feature](https://github.com/monero-project/monero/pull/6418/files) (just merged a couple of weeks ago), and subscribe to `txpool_add`.

ZeroMQ has binding for most popular languages, so you could easily write a script in Python, NodeJS, or whatever language you prefer.

**Comment by wojtasss** [comment]
@ndorf sounds great! Need I use bindings for ZeroMQ for particular language or can just connect websockets to this ZeroMQ on Monero full node?

**Comment by ndorf** [comment]
WebSockets are unrelated to ZeroMQ. You would need to use a ZeroMQ binding. It's not hard; here's a simplistic example in Python:

```
import zmq

ctx = zmq.Context()
sock = ctx.socket(zmq.SUB)

sock.connect("ipc:///var/run/monero/zmqpub.sock")
print("Connected to daemon")

sock.subscribe("json-minimal-txpool_add")
print("Subscribed")

while True:
    msg = sock.recv()
    print("Received message: %s" % msg)
```

**Comment by wojtasss** [comment]
Ok, I just was wonderng if I don't need add more layers to my project (game engine Godot with GDscript language that supports websockets) but probably I will stay with just download current txpool from monerod :) anyway thanks!

**Comment by moneromooo-monero** [comment]
--tx-notify tells you about txes in the pool (if they are for you).

**Comment by wojtasss** [comment]
@moneromooo-monero yup I know that, but need notify when transaction arrived to txpool so want to get info about all transactions in txpool not only mine
